### PR TITLE
Reworked categories

### PR DESCRIPTION
**Pull request by armored-guitar**

### CODE REVIEW

File: oml/datasets/retrieval.py [hunk 1]
```diff
@@ -71,6 +74,9 @@ def __init__(
             df["y_1"] = None
             df["y_2"] = None
 
+        if "category" not in df.columns:
+            df["category"] = "OVERALL"
```
Comment (DaloroAT) [inline oml/datasets/retrieval.py:78]: Need copy before modifying?

**Review by DaloroAT — APPROVED** [review]

File: oml/datasets/retrieval.py (hunk 1 above)
Comment (AlekseySh) [reply oml/datasets/retrieval.py:78]: added copy

File: examples/inshop/convert_inshop.py [hunk 2]
```diff
@@ -131,12 +117,27 @@ def build_inshop_df(
...
 
-    df = df[
-        ["label", "path", "split", "is_query", "is_gallery", "x_1", "x_2", "y_1", "y_2", "category", "category_name"]
-    ]
+    # check stat
+    assert df["path"].nunique() == len(df) == 52712
+    assert df["label"].nunique() == 7982
+    assert set(df["label"].astype(int).tolist()) == set(list(range(1, 7982 + 1)))
+
+    mask_non_single_images = df.groupby("label").label.transform("count") > 1
```
Comment (DaloroAT) [inline examples/inshop/convert_inshop.py:129]: this step have to launch as last step

Dropping bad boxes may lead to label with single instance

File: oml/lightning/callbacks/metric.py [hunk 3]
```diff
@@ -73,7 +73,9 @@ def on_validation_epoch_end(self, trainer: pl.Trainer, pl_module: pl.LightningMo
 
     def calc_and_log_metrics(self, pl_module: pl.LightningModule) -> None:
         metrics = self.metric.compute_metrics()
-        metrics = flatten_dict(metrics, sep="/")
+        # In Tensorboard/Neptune we only log metrics for the main category
+        metrics = {self.metric.overall_categories_key: metrics[self.metric.overall_categories_key]}
```
Comment (DaloroAT) [inline oml/lightning/callbacks/metric.py:77]: pray remove it 
Or modify init in callback for this option

File: oml/metrics/triplets.py [hunk 4]
```diff
@@ -9,50 +9,34 @@
...
         self.embeddings_key = embeddings_key
         self.categories_key = categories_key
-        self.categories_names_mapping = categories_names_mapping
 
-        self.avg_online = OnlineAvgDict()
+        self.avg_online = defaultdict(OnlineAvgDict)  # type: ignore
 
     def setup(self, *args: Any, **kwargs: Any) -> None:  # type: ignore
-        self.avg_online = OnlineAvgDict()
+        pass
```
Comment (DaloroAT) [inline oml/metrics/triplets.py:27]: add `self.avg_online = defaultdict(OnlineAvgDict) # type: ignore`

We need to flush avg for next epoch

**Review by DaloroAT — APPROVED** [review]

File: examples/inshop/convert_inshop.py (hunk 2 above)
Comment (AlekseySh) [reply examples/inshop/convert_inshop.py:129]: done

File: oml/lightning/callbacks/metric.py (hunk 3 above)
Comment (AlekseySh) [reply oml/lightning/callbacks/metric.py]: ok

**Pull request merged**